### ISSUE TITLE
Validating if one of nested subcommands exist

### ISSUE THREAD

**Issue by vijaydaiict** [scallop/scallop#197]
Is there a way to validate if the nested subcommands?? 

MainConf -> subcommand1 -> subcommand2. If the subcommnd1 is provided then is there a way to write validation that one of the valid subcommnd2 exists. 

For the MainConf.requireSubCommand works only for the subCommand1. 

If we use  subcommand1.requireSubCommand then it fails with exception that Subcommand cannot be called on the SubCommand object

**Comment by Rogach** [comment]
I released version `3.3.2` that allows you to validate nested subcommands the same way as top-level subcomands. Here's an example of the code:

```scala
object Conf extends ScallopConf(Seq("fruit", "pick", "-c", "42")) {
  val fruit = new Subcommand("fruit") {
    val pick = new Subcommand("pick") {
      val count = opt[Int]()
    }
    addSubcommand(pick)
    requireSubcommand()
  }
  addSubcommand(fruit)
  requireSubcommand()

  verify()
}
```

**Comment by vijaydaiict** [comment]
Gr8 thanks
